iothub_client/src/uamqp_messaging.c

```diff
@@ -430,46 +434,102 @@ int message_create_uamqp_encoding_from_iothub_message(IOTHUB_MESSAGE_HANDLE mess
 
 static int readPropertiesFromuAMQPMessage(IOTHUB_MESSAGE_HANDLE iothub_message_handle, MESSAGE_HANDLE uamqp_message)
```

Comment:
>readPropertiesFromuAMQPMessage [](start = 11, length = 30)

This fcn is +250 lines which is getting unweildy.  

Could we please have a helper function that would do something like a GetStringVariant (please don't call it a Variant though :)) where you'd do llike GetStringVarianr(property_name, string_buffer, string_buffer_length) and then it'd do the str/long/uuid determination.

Reply:
You are correct, but this is an improvement. It will be done in another time.
I'll go ahead with the check in, we need to kick the release today.